rust/sbp/src/sbp_string.rs

```diff
@@ -37,6 +42,98 @@ where
     }
 }
 
+impl<T: AsRef<[u8]>> SbpString<T, Unterminated> {
+    /// Checked unterminated SbpString builder, verify last byte is non null
+    pub fn unterminated(data: T) -> Result<Self, UnterminatedError> {
+        match data.as_ref().last() {
+            Some(l) => {
+                if l != &0 {
+                    Ok(Self::new(data))
+                } else {
+                    Err(UnterminatedError)
+                }
+            }
+            None => Err(UnterminatedError),
+        }
+    }
+}
+
+impl<T: AsRef<[u8]>> SbpString<T, NullTerminated> {
+    pub fn null_terminated(data: T) -> Result<Self, NullTerminatedError> {
+        match data.as_ref().last() {
+            Some(l) => {
+                if l == &0 {
+                    Ok(Self::new(data))
+                } else {
+                    Err(NullTerminatedError)
+                }
+            }
+            None => Err(NullTerminatedError),
+        }
+    }
+}
+
+// pub type Parts<'a> = std::slice::Split<'a, u8, fn(&u8) -> bool>;
+
+impl SbpString<Vec<u8>, Multipart> {
+    pub fn multipart(data: impl Into<Vec<u8>>) -> Result<Self, MultipartError> {
+        let vec = data.into();
+        if let [.., one] = vec.as_slice() {
```

Comment:
is this basically the idea?

- we don't actually check if the next is non null so a multipart could be a double null... - should we handle that case as well or assume it should be multipart